### PR TITLE
remove optional type hint

### LINKED ISSUE #1939: [CurrentDayConditions] Fix typing in `CurrentDayConditions`
The type hints in `CurrentDayConditions` should be corrected.

## Status

Several attributes in `CurrentDayConditions` are typed as `Optional[...]` when they should not be.

## Intention

Have the type hints accurately reflect which types we expect variables to have in RuFaS.

## Request

Please change the type hints for `incoming_light`, the min, max, and mean temps, and the avg annual air temp. `daylength` should stay optional, because it is set to `None` when no latitude is provided to the methods in `Weather` that collect `CurrentDayConditions`.

### PR DESCRIPTION
**Pull request by matthew7838**
Remove the optional type hint from current_day_condition.py.
### Context

Issue(s) closed by this pull request: closes #1939 

### What

Several attributes in CurrentDayConditions are typed as Optional[...] when they should not be. The annual_mean_air_temperature will remain optional with the default None value because it will be later calculated and assigned in weather.py after all the mean air temperature has been loaded to initiate each day's current_day_condition. Therefore, placeholders are needed. Let me know if it's better to remove the optional but keep the default `None`.

### Why

The type hints should accurately reflect which types we expect variables to have in RuFaS.

### How

Remov the type hints.

### Test plan

- Updated and run the test suite.
- Run with default, barnyard, and available simulation metadata, and nothing is broken.

### CODE REVIEW

File: input/data/feed/default_feed.json [hunk 1]
```diff
@@ -8,28 +8,28 @@
         44,
         51,
         110,
-        167, 
+        167,
```
Comment (ew3361zh) [inline input/data/feed/default_feed.json:11]: Not clear why the changes to default_feed.json are in this PR.

**Review by ew3361zh — APPROVED** [review]
LGTM, nice work, Matthew.

Only question is about default_feed.json changes which don't seem like they were meant to be a part of this PR so they should either be removed or accounted for in the PR description.

I did see a few errors that cropped up that you mentioned in our worktogether today - are you writing up issues for those? The simulation failed for 3 of the metadatas: swat, barnyard, and testing.

File: input/data/feed/default_feed.json (hunk 1 above)
Comment (matthew7838) [reply input/data/feed/default_feed.json:11]: Seems like some auto-formatting.

Comment (ew3361zh) [reply input/data/feed/default_feed.json:11]: Oh it does look like that. Weird that it would decide to format this file at this time.

**Review by Isgotkowitz — APPROVED** [review]
LGTM!

**Comment by matthew7838** [comment]
> LGTM, nice work, Matthew.
> 
> Only question is about default_feed.json changes which don't seem like they were meant to be a part of this PR so they should either be removed or accounted for in the PR description.
> 
> I did see a few errors that cropped up that you mentioned in our worktogether today - are you writing up issues for those? The simulation failed for 3 of the metadatas: swat, barnyard, and testing.

I will write issues for that.
